portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,23 @@ async function getMessageFromServer() {
   const quote = await response.text();
   document.getElementById('message-container').innerHTML = quote;
 }
+
+function login(url){
+  window.location = url;
+}
+
+function logout(url){
+  window.location = url;
+}
+
+async function checkLoginStatus(){
+  const response = await fetch('/login');
+  const status = await response.text();
```

Comment:
this could return JSON with the login status (whether it says "Login or Logout") plus the URL you need